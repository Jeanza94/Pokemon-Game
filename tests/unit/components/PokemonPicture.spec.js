import { shallowMount } from "@vue/test-utils";
import PokemonPicture from '@/components/PokemonPicture'

describe('Purebas en el componente PokemonOptions', () => { 

    test('debe de hacer match con el snapshot', () => { 

        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: false
            }
        });

        expect(wrapper.html()).toMatchSnapshot();

    });

    test('debe de mostrar la imagen oculta y el pokemon 100', () => { 
        
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        });

        const img = wrapper.find('img');
        const {src, alt, class:clase} = img.attributes();
        
        expect(src).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')
        expect(clase).toBe('hidden-pokemon')

    });

    test('debe de mostrar el pokemon si showPokemon esta en true', () => { 
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: true
            }
        });

        const img = wrapper.find('img');
        const { src, alt, class: clase } = img.attributes();

        expect(clase).toBe('fade-in')
    });

});