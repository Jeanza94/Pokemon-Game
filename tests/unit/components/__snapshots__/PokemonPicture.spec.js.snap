// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Purebas en el componente PokemonOptions debe de hacer match con el snapshot 1`] = `<div class="pokemon-contain"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon" class="hidden-pokemon"></div>`;
